templates/tfengine/recipes/resources.hcl

```diff
@@ -434,6 +448,74 @@ schema = {
         }
       }
     }
+    dns_zones = {
+      description = "https://github.com/terraform-google-modules/terraform-google-cloud-dns"
+      type        = "array"
+      items = {
+        type = "object"
+        additionalProperties = false
+        required = [
+          "name",
+          "domain",
+          "type",
+        ]
+        properties = {
+          name = {
+            description = "Name of DNS zone."
+          }
+          resource_name = {
+            description = <<EOF
+              Override for Terraform resource name. If unset, defaults to normalized name.
+              Normalization will make all characters alphanumeric with underscores.
+            EOF
+            type        = "string"
+          }
+          domain = {
+            description = "Domain of DNS zone. Must end with period."
```

Comment:
You should add a regex to verify that it ends with a period.

Reply:
Done.